packages/eslint-plugin/tests/rules/no-misused-spread.test.ts

```diff
@@ -0,0 +1,110 @@
+import { RuleTester } from '@typescript-eslint/rule-tester';
+
+import rule from '../../src/rules/no-misused-spread';
+import { getFixturesRootDir } from '../RuleTester';
+
+const rootDir = getFixturesRootDir();
+const ruleTester = new RuleTester({
+  parser: '@typescript-eslint/parser',
+  parserOptions: {
+    tsconfigRootDir: rootDir,
+    project: './tsconfig.json',
+  },
+});
+
+ruleTester.run('no-misused-spread', rule, {
+  valid: [
+    `
+const a = () => ({ value: 33 });
+const b = {
+  ...a(),
+  name: 'name',
+};
+    `,
+    `
+const a = (x: number) => ({ value: x });
+const b = {
+  ...a(42),
+  name: 'name',
+};
+    `,
+    `
+interface FuncWithProps {
+  property?: string;
+  (): number;
+}
+const funcWithProps: FuncWithProps = () => 1;
+funcWithProps.property = 'foo';
+const spreadFuncWithProps = { ...funcWithProps };
+    `,
+    `
+type FuncWithProps = {
+  property?: string;
+  (): number;
+};
+const funcWithProps: FuncWithProps = () => 1;
+funcWithProps.property = 'foo';
+const spreadFuncWithProps = { ...funcWithProps };
+    `,
+    `
+const a = { value: 33 };
+const b = {
+  ...a,
+  name: 'name',
+};
+    `,
+    `
+const a = {};
+const b = {
+  ...a,
+  name: 'name',
+};
+    `,
+    `
+const a = () => ({ value: 33 });
+const b = ({ value: number }) => ({ value: value });
+b({ ...a() });
+    `,
+    `
+const a = () => ({ value: 33 });
+const b = ({ value: number }) => ({ value: value });
+    `,
```

Comment:
I think it is a copy/paste from the previous in order to test another case, but honestly I don't remember since last time I worke on this PR.
I remove it.